linode/manager
# upcoming: [M3-7354] DC Get Well - Disabled regions

## Description 📝
This PR implements the UI for disabled regions in the `<RegionSelect />`. We originally had hardcoded "fake" regions  to represent disabled menu items. DC Get Well now implements a new endpoint (`account/availability`) which represents an array of DCs (or regions) unavailable to the user. 

The UI is essentially the same (disabled item plus tooltip with link to a static info page - link TBD).

Right now this PR does not implement the behavior in all `RegionSelect` instances to keep the PR small and focus on pure logic implementation. (see testing steps)

## Changes  🔄
- Implement new `account/availability` request (mocked for now) and underlying logic to determine the region's availability to a given user.
- Implement a new prop to pass the current capability to the region select
- remove all logic pertaining to fake regions 🎉 
- cleanup and enhanced test suites

## Preview 📷

![Screenshot 2023-11-21 at 13 35 40](https://github.com/linode/manager/assets/130582365/34f08b4d-6b2f-4248-a578-f484fa2bdd6e)

## How to test 🧪

### Prerequisites
- Pull code locally
- Turn on the DC Get Well feature flag in dev settings
- Turn on MSW

### Verification steps 
- You can verify the behavior by adding the current capability prop (you can try for various create flows)

ex: add `currentCapability="Block Storage"` to the [VolumeCreate region select](https://github.com/linode/manager/blob/743c345141708cfcb851f5f16d00f1ed8c0609b0/packages/manager/src/features/Volumes/VolumeCreate.tsx#L293) 

👉  verify that Atlanta and Tokyo are disabled items

## As an Author I have considered 🤔

*Check all that apply*
